kaggle_environments/envs/lux_ai_2022/luxai2022/env.py

```diff
@@ -472,7 +498,7 @@ def step(self, actions):
                     valid_acts, err_reason = self.action_space(agent).contains(unit_actions)
                     if not valid_acts:
                         failed_agents[agent] = True
-                        raise ValueError(f"{self.state.teams[agent]} Inappropriate action given. {err_reason}")
+                        self._log(f"{self.state.teams[agent]} Inappropriate action given. {err_reason}")
```

Comment:
Why not throw here? are we allowing people to send invalid actions?

Reply:
This causes the environment to error if we threw. Then the server wouldn't run matches?

Reply:
right, good call. but should we kill the unit that sends the bad action or something?

Reply:
The game just ends right away and we mark them as the loser.

Reply:
Or do you mean do you want to mark the agent as "failed" status instead of "done" status. By default I always use "done"